main.js

```diff
@@ -73,6 +78,11 @@ async function downloadCLI(version, platform, arch) {
   }
 }
 
+// Verifies the download
+async function verifyDownload(downloadPath) {
+    await exec.exec('gh', ['attestation', 'verify', downloadPath, '--repo', 'UpCloudLtd/upcloud-cli']);
```

Comment:
I'm not sure if there could be an environment where `gh` is not installed or authenticated.

Reply:
Custom runners at least.

Reply:
[Official runner VM images](https://github.com/actions/runner-images/blob/main/images/ubuntu/templates/ubuntu-24.04.pkr.hcl#L289) by Github seem to install the CLI. But [containerised runners don't](https://github.com/actions/runner/blob/main/images/DockerfileArtesanal). Probably the same for artesanal runners.

Reply:
I guess we could add a `verify-attestation` boolean to turn this off (default on).

Reply:
Added, squashed, rebased, lifting draft.